File: src/test/java/test/GetUserLoginHistoryTest.java
import cn.authing.sdk.java.client.ManagementClient;
import cn.authing.sdk.java.dto.*;
import cn.authing.sdk.java.model.ManagementClientOptions;
import cn.authing.sdk.java.util.JsonUtils;

import java.util.Collections;



public class GetUserLoginHistoryTest {

    private static final String ACCESS_KEY_ID = "YOUR_ACCESS_KEY_ID";
    private static final String ACCESS_KEY_SECRET = "YOUR_ACCESS_KEY_SECRET";


    public static void main(String[] args) throws Throwable {
        ManagementClientOptions clientOptions = new ManagementClientOptions(ACCESS_KEY_ID, ACCESS_KEY_SECRET);
        ManagementClient client = new ManagementClient(clientOptions);

        GetUserLoginHistoryDto request = new GetUserLoginHistoryDto();
        request.setUserId("userId_3143");
        request.setUserIdType("userIdType_7931");
        request.setAppId("appId_7326");
        request.setClientIp("clientIp_156");
        request.setStart(0);
        request.setEnd(0);
        request.setPage(0);
        request.setLimit(0);

        UserLoginHistoryPaginatedRespDto response = client.getUserLoginHistory(request);
        System.out.println(JsonUtils.serialize(response));
    }

}

File: src/test/go/test/GetUserLoginHistoryTest.go
package main

import (
"authing-go-sdk/client"
"authing-go-sdk/dto"
"fmt"
)

func main() {
    options := client.ManagementClientOptions{
        AccessKeyId:     "AUTHING_USERPOOL_ID",
        AccessKeySecret: "AUTHING_USERPOOL_SECRET",
    }
    var err error
    client, err := client.NewClient(&options)
    if err != nil {
        panic(err)
    }
    request := dto.GetUserLoginHistoryDto{
        UserId: "userId_1177",
        UserIdType: "userIdType_2994",
        AppId: "appId_1866",
        ClientIp: "clientIp_7180",
        Start: 0,
        End: 0,
        Page: 0,
        Limit: 0,
    }
    response := client.GetUserLoginHistory(&request)
    fmt.Println(response)
}

